Sources/Datadog/RUM/RUMVitals/VitalMemoryReader.swift

```diff
@@ -0,0 +1,33 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+internal class VitalMemoryReader: VitalReader {
+    static let TASK_VM_INFO_COUNT = MemoryLayout<task_vm_info>.size / MemoryLayout<natural_t>.size
+
+    func readVitalData() -> Double? {
+        var vmInfo = task_vm_info()
+        var vmInfoSize = mach_msg_type_size_t(VitalMemoryReader.TASK_VM_INFO_COUNT)
+
+        let kern: kern_return_t = withUnsafeMutablePointer(to: &vmInfo) {
+            $0.withMemoryRebound(to: integer_t.self, capacity: 1) {
+                task_info(
+                    mach_task_self_,
+                    task_flavor_t(TASK_VM_INFO),
+                    $0,
+                    &vmInfoSize
+                )
+            }
+        }
+
+        if kern == KERN_SUCCESS {
+            return Double(vmInfo.resident_size)
```

Comment:
does that give how much memory is being used only? does that provide value to the user? shouldn't we provide more context such as max memory/memory utilization % ?

Reply:
We can look into knowing how much the device is capable in a future version, but for now just the size used should already give some info